apps/arweave/src/ar_http_iface_middleware.erl

```diff
@@ -504,6 +507,66 @@ handle(<<"POST">>, [<<"block2">>], Req, Pid) ->
 	erlang:put(post_block2, true),
 	post_block(request, {Req, Pid, binary}, erlang:timestamp());
 
+%% Accept a (partial) solution from a pool proxy or a CM node and validate it.
```

Comment:
I think I'm getting a little lost about all the different entities in the system. I see:
- Pool Server
- Pool Proxy
- Standalone Pool Client
- CM Exit Node Pool Client

Is that all of them? Any chance you can put an ascii diagram in somewhere (maybe `ar_pool.erl`) to clarify the communication? Is it something like

```

                                +---> Standalone Pool Client
                                |
Pool Server <--> Pool Proxy <---+
                                |
                                +---> CM Exit Node Pool Client <--> CM Miner
```

Reply:
And actually it might also help to diagram how the partial solution is shared. It sounds like there are 2 different flows:

1. No coordinated mining:
```
Standalone Pool Client -- partial solution -> Pool Proxy -- partial solution -> Pool Sever
```

2. Coordinated mining:
```
CM Miner -- partial solution -> CM Exit Node Pool Client -- partial solution -> Pool Proxy -- partial solution -> Pool Server
```

Reply:
Also (and this may just be my ignorance) it would be great to define "partial solution" somewhere. As I understand it, it's a valid solution that satisfies a lower difficulty threshold than is required to publish a block. *But* it can also satisfy the higher difficulty threshold. So the set of all partial solutions includes the set of all full solutions - right?

Reply:
A "normal" solution may not satisfy the difficulty threshold either (e.g., because the difficulty has just jumped up) so the main difference is in how we use them - partial solutions serve to prove the ongoing mining activity. Yes, the set of all solutions of the blocks produced by the pool come from "partial" solutions technically (they go through the same pipeline).